backend/layers/persistence/persistence_mock.py

```diff
@@ -36,7 +43,7 @@ class DatabaseProviderMock(DatabaseProviderInterface):
     """
 
     # A mapping between canonical collection ids and collection versions.
-    collections: Dict[str, str]
+    collections: Dict[str, CanonicalCollectionPrivate]
```

Comment:
`Private` here refers to the fact that this class belongs to the persistence mock and should not leak outside. Any suggestions for a better name? 😄

Reply:
Ah! `_CanonicalCollection`

Reply:
😬 ugly, but pythonic I guess. Going for it!

Reply:
> Now that we have `CollectionVersion.canonical_collection`, could this just be `Dict[str, CollectionVersion]`?

No, because that would not contain the new fields (`published_at` and `tombstoned`).

Reply:
`collections.get(cc_id).canonical_collection.originally_published_at` wouldn't work?

Reply:
> `collections.get(cc_id).canonical_collection.originally_published_at` wouldn't work?

(see my edit above)